src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java

```diff
@@ -583,92 +372,44 @@ private Future<ReceiptStatus> calculatePoLineReceiptStatus(List<Piece> byPoLine,
-    } else {
-      status.setType(ProcessingStatus.Type.FAILURE);
-      status.setError(error);
-      resultCounts.merge(ProcessingStatus.Type.FAILURE.toString(), 1, Integer::sum);
-    }
-
-    ReceivingItemResult itemResult = new ReceivingItemResult();
-    itemResult.setPieceId(pieceId);
-    itemResult.setProcessingStatus(status);
-    result.getReceivingItemResults().add(itemResult);
-  }
 
   /**
    * Filter by locationId presence for items/pieces related to POLine.
    *
    * @return {@link Future} which holds map with PO line id as key
-   *         and list of corresponding pieces as value
+   * and list of corresponding pieces as value
    */
   Future<Map<String, List<Piece>>> filterMissingLocations(Map<String, List<Piece>> piecesRecords, RequestContext requestContext) {
     return getPoLines(StreamEx.ofKeys(piecesRecords).toList(), requestContext)
       .map(poLines -> {
-        for(PoLine poLine : poLines) {
+        for (PoLine poLine : poLines) {
           piecesRecords.get(poLine.getId()).removeIf(piece -> isMissingLocation(poLine, piece));
         }
         return piecesRecords;
       });
   }
 
   private boolean isMissingLocation(PoLine poLine, Piece piece) {
-    // Check if locationId doesn't presented in piece from request and retrieved from storage
-    // Corresponding piece from collection
-    if ((getLocationId(piece) == null && getHoldingId(piece) == null) && !isRevertToOnOrder(piece)) {
-      if (piece.getFormat() == Piece.Format.ELECTRONIC) {
-        // Check E-Resource
-        if (poLine.getEresource() != null && poLine.getEresource().getCreateInventory() != Eresource.CreateInventory.NONE
-          && poLine.getEresource().getCreateInventory() != Eresource.CreateInventory.INSTANCE) {
-          addError(poLine.getId(), piece.getId(), LOC_NOT_PROVIDED.toError());
-          return true;
-        }
-      } else {
-        // Check Physical
-        if (poLine.getPhysical() != null && poLine.getPhysical().getCreateInventory() != Physical.CreateInventory.NONE
-          && poLine.getPhysical().getCreateInventory() != Physical.CreateInventory.INSTANCE) {
-          addError(poLine.getId(), piece.getId(), LOC_NOT_PROVIDED.toError());
-          return true;
-        }
+    if (getLocationId(piece) != null || getHoldingId(piece) != null || isRevertToOnOrder(piece)) {
+      return false;
+    }
+    if (piece.getFormat() == Piece.Format.ELECTRONIC) {
+      // Check E-Resource
+      if (poLine.getEresource() != null && poLine.getEresource().getCreateInventory() != Eresource.CreateInventory.NONE
+        && poLine.getEresource().getCreateInventory() != Eresource.CreateInventory.INSTANCE) {
+        addError(poLine.getId(), piece.getId(), LOC_NOT_PROVIDED.toError());
+        return true;
+      }
+    } else {
+      // Check Physical
+      if (poLine.getPhysical() != null && poLine.getPhysical().getCreateInventory() != Physical.CreateInventory.NONE
```

Comment:
we can use else if to simply this block

Reply:
Change from first commit - I understand many things can be simplified but we can't do all this in scope of one commit

Reply:
Ok, sorry about that, You can ignore the review is not related to in scope of the ticket and I will resolve them, I am reviewing as much as possible.

Reply:
Thank you!